src/liboslexec/runtimeoptimize.cpp

```diff
@@ -2121,6 +2122,26 @@ RuntimeOptimizer::optimize_ops (int beginop, int endop,
         // Nothing below here to do for no-ops, take early out.
         if (op->opname() == u_nop)
             continue;
+        // If an exit or return was reached, kill the remaining instructions
+        if (optimize() && (op->opname() == u_return || op->opname() == u_exit)) {
+            ASSERT(skipops == 0 && "Unexpected skipops");
+            if (m_in_conditional[opnum]) {
+                // Nop the remaining instructions in the same basic block
+                while (next_block_instruction(opnum)) {
+                    ++skipops;
+                    op = &inst()->ops()[++opnum];
+                    turn_into_nop (*op, "eliminated unreachable op");
+                }
+            } else {
+                // Nop the remaining instructions, preserving the end-op
```

Comment:
Let's update this comment to be a little more explanatory:

    // Since we hit an 'exit' in unconditionally-executed code, it is safe to
    // eliminate all the remaining code up to final exit op.

I think this part is correct, **if** you fix it so that this is happening only for `exit` and not for `return`.